src/tree-manager/getPartialState.js

```diff
@@ -0,0 +1,8 @@
+import partial from 'array.partial'
+
+const identity = c => c
+
+export default (node, childProp = 'children', childSelector = identity) => {
+  if (partial(node[childProp], c => childSelector(c).checked)) return true
```

Comment:
```
return (partial(node[childProp], c => childSelector(c).checked))  ||
  node[childProp].some(c => childSelector(c).partial)
```

Reply:
I think I had it that way too. Thing is, it creates issues. 

Here's the logic this block is trying to execute:
- `array.partial` returns true _only & only if_ at least _one_ child is selected. If all/none are selected, it returns false. In other words, a parent is partial if one or more (but not 0 or all) children are checked. So if this condition holds, return true and stop further inspection.

- Say above check failed, then check if one or more nodes are partial (can happen if the nodes have grandchildren). If so, then this node is also partial. 

- The last case where all children are checked already gets handled properly (`setNodeCheckedState` and `toggleChildren` take care of it) so we can live with `array.some` in the last check.

Try using your logic and run the tests. You'd see the problem. It's also what @toofff's [comment](https://github.com/dowjones/react-dropdown-tree-select/pull/73#issuecomment-376768214) pointed out. 

It's also possible I've grossly overlooked stuff in my midnight stupor, so let me know if you find anything.

Reply:
I understand your logic and your explanation but I don't understand why the `||` fails. But it's not a big deal, you can keep it that way.